go/ql/test/library-tests/semmle/go/frameworks/Fasthttp/fasthttp.go

```diff
@@ -0,0 +1,234 @@
+		requestCtx.URI().FullURI()         // $ UntrustedFlowSource="call to FullURI"
+		requestCtx.URI().LastPathSegment() // $ UntrustedFlowSource="call to LastPathSegment"
+		requestCtx.URI().QueryString()     // $ UntrustedFlowSource="call to QueryString"
+		requestCtx.URI().String()          // $ UntrustedFlowSource="call to String"
+
+		//or requestCtx.PostArgs()
+		requestCtx.URI().QueryArgs().Peek("arg1")                   // $ UntrustedFlowSource="call to Peek"
+		requestCtx.URI().QueryArgs().PeekBytes([]byte("arg1"))      // $ UntrustedFlowSource="call to PeekBytes"
+		requestCtx.URI().QueryArgs().PeekMulti("arg1")              // $ UntrustedFlowSource="call to PeekMulti"
+		requestCtx.URI().QueryArgs().PeekMultiBytes([]byte("arg1")) // $ UntrustedFlowSource="call to PeekMultiBytes"
+		requestCtx.URI().QueryArgs().QueryString()                  // $ UntrustedFlowSource="call to QueryString"
+		requestCtx.URI().QueryArgs().String()                       // $ UntrustedFlowSource="call to String"
+		requestCtx.String()                                         // $ UntrustedFlowSource="call to String"
+		// not sure what is the best way to write query for following
+		//requestCtx.URI().QueryArgs().VisitAll(type func(,))
+
+		requestCtx.Path() // $ UntrustedFlowSource="call to Path"
+		// multipart.Form is already implemented
+		// requestCtx.FormFile("FileName")
+		// requestCtx.FormValue("ValueName")
+		requestCtx.Referer()           // $ UntrustedFlowSource="call to Referer"
+		requestCtx.PostBody()          // $ UntrustedFlowSource="call to PostBody"
+		requestCtx.RequestBodyStream() // $ UntrustedFlowSource="call to RequestBodyStream"
+		requestCtx.RequestURI()        // $ UntrustedFlowSource="call to RequestURI"
+		requestCtx.UserAgent()         // $ UntrustedFlowSource="call to UserAgent"
+		requestCtx.Host()              // $ UntrustedFlowSource="call to Host"
+
+		requestCtx.Request.Host()             // $ UntrustedFlowSource="call to Host"
+		requestCtx.Request.Body()             // $ UntrustedFlowSource="call to Body"
+		requestCtx.Request.RequestURI()       // $ UntrustedFlowSource="call to RequestURI"
+		requestCtx.Request.BodyGunzip()       // $ UntrustedFlowSource="call to BodyGunzip"
+		requestCtx.Request.BodyInflate()      // $ UntrustedFlowSource="call to BodyInflate"
+		requestCtx.Request.BodyUnbrotli()     // $ UntrustedFlowSource="call to BodyUnbrotli"
+		requestCtx.Request.BodyStream()       // $ UntrustedFlowSource="call to BodyStream"
+		requestCtx.Request.BodyUncompressed() // $ UntrustedFlowSource="call to BodyUncompressed"
+		requestCtx.Request.ReadBody(dstReader, 100, 1000)
+		requestCtx.Request.ReadLimitBody(dstReader, 100)
+		requestCtx.Request.ContinueReadBodyStream(dstReader, 100, true)
+		requestCtx.Request.ContinueReadBody(dstReader, 100)
+		// not sure what is the best way to write query for following
+		//requestCtx.Request.Header.VisitAllCookie()
```

Comment:
I am looking into whether we can model this. (Should we also model `RequestHeader.VisitAll`, `RequestHeader.VisitAllInOrder`, `RequestHeader.VisitAllTrailer`? I assume not, since you didn't mention them.)

Reply:
See comment above.